### ISSUE TITLE
Core engine 

### ISSUE THREAD

**Issue by roger-lo** [roger-lo/filed#6]
Couldn't think of a better name. I imagine this being the main object. 

It will be responsible for:
- Instantiating the different plugins / filters based on the config file. 
- Parsing command arguments

**Comment by ejwmoreau** [comment]
Yeah, I think that's a good start. If any particular part of the Core Engine gets too big, then we could refactor it out into a separate class.
